client/src/App.tsx

```diff
@@ -46,6 +49,18 @@ const router = createBrowserRouter([
         path: "challenge/:id",
         element: <Challenge />,
       },
+      {
+        path: "solutions",
+        element: <Solutions />,
+      },
+      {
+        path: "solution/:id",
```

Comment:
So `solution/:id` is for viewing a solution given its id and` solution/post/:id` is to edit existing solution?